developer_manual/digging_deeper/text2image.rst

```diff
@@ -0,0 +1,187 @@
+.. _text2image:
+
+=============
+Text-To-Image
+=============
+
+.. versionadded:: 28
+
+Nextcloud offers a **Text-To-Image** API. The overall idea is that there is a central OCP API that apps can use to prompt tasks to latent diffusion AI models and similar image generation tools. To be technology agnostic any app can provide this functionality by registering a Text-To-Image provider.
+
+Consuming the Text-To-Image API
+-------------------------------
+
+To consume the Text-To-Image API, you will need to :ref:`inject<dependency-injection>` ``\OCP\TextToImage\IManager``. This manager offers the following methods:
+
+ * ``hasProviders()`` This method returns a boolean which indicates if any providers have been registered. If this is false you cannot use the image generation feature.
+ * ``runTask(Task $task)`` This method provides the actual functionality. The task is defined using the Task class. This method runs the task synchronously, so depending on the implementation it is uncertain how long it will take (between 3s and several hours).
+ * ``scheduleTask(Task $task)`` This method also runs a task, but asynchronously in a background job. The task is defined using the Task class.
+ * ``runOrScheduleTask(Task $task)`` This method also runs a task, but fist checks the expected runtime of the provider to be used. If the runtime fits inside the available processing time for the current request the task is run synchronously, otherwise it is scheduled as a background job. The task is defined using the Task class.
```

Comment:
How to handle the outcome of runOrScheduleTask is somewhat unclear based on the docs/dockblocks as it requires different logic based on whether the task was run or scheduled. Would it make sense to provide a more info and/or a sample code snippet of how to deal with outputs of runTask/scheduleTask/runOrScheduleTask?

(Minor typo on this line: "fist")

Reply:
Yeah, good idea!